edb/edgeql/compiler/conflicts.py

```diff
@@ -800,6 +800,11 @@ def compile_inheritance_conflict_checks(
 
     conflicters = []
     for subject_stype, anc_type, ir in modified_ancestors:
+
+        # don't enforce any constraints for abstract object type
```

Comment:
I think this is wrong. I believe that skipping this check means that we won't generate a constraint check against an abstract common ancestor, so we might miss enforcing a constraint.

I suspected that was the case, so I tried a variant of your test case that made the property single, and it does incorrectly succeed.

I'm not sure why the multi version works.

Reply:
I pushed the test case I wrote

Reply:
I was wrong that this was wrong. I had mixed up which type was which, and thought that it was being skipped when the source of the constraint was abstract. This code should be right.

But it was good that I was wrong about this and decided to poke at it, because the test is of course still failing!